src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -236,6 +237,204 @@ public static void Main()
             }
         }
 
+        [Theory]
+        [InlineData("public int M() { return 1; }", "public int M() { return 2; }", true)]
+        [InlineData("public int M() { return 1; }", "public int M() { error(); }", true)]
+        [InlineData("", "private void M() { }", false)] // Should be true. See follow-up issue https://github.com/dotnet/roslyn/issues/17612
+        [InlineData("internal void M() { }", "", false)]
+        [InlineData("public struct S { private int i; }", "public struct S { }", false)]
+        public void RefAssembly_InvariantToSomeChanges(string change1, string change2, bool expectMatch)
+        {
+            string sourceTemplate = @"
+public class C
+{
+    CHANGE
+}
+";
+            string name = GetUniqueName();
+            string source1 = sourceTemplate.Replace("CHANGE", change1);
+            CSharpCompilation comp1 = CreateCompilationWithMscorlib(Parse(source1),
+                options: TestOptions.DebugDll.WithDeterministic(true), assemblyName: name);
+
+            ImmutableArray<byte> image1;
+
+            using (var output = new MemoryStream())
+            {
+                image1 = comp1.EmitToArray(EmitOptions.Default.WithEmitMetadataOnly(true));
+            }
+
+            var source2 = sourceTemplate.Replace("CHANGE", change2);
+            Compilation comp2 = CreateCompilationWithMscorlib(Parse(source2),
+                options: TestOptions.DebugDll.WithDeterministic(true), assemblyName: name);
+
+            ImmutableArray<byte> image2;
+
+            using (var output = new MemoryStream())
+            {
+                image2 = comp2.EmitToArray(EmitOptions.Default.WithEmitMetadataOnly(true));
+            }
+
+            if (expectMatch)
+            {
+                AssertEx.Equal(image1, image2);
+            }
+            else
+            {
+                AssertEx.NotEqual(image1, image2);
+            }
+        }
+
+        [Theory]
+        [InlineData("public int M() { error(); }", true)]
+        [InlineData("public int M() { error() }", false)] // Should be true. See follow-up issue https://github.com/dotnet/roslyn/issues/17612
+        [InlineData("public Error M() { return null; }", false)] // Should be true. See follow-up issue https://github.com/dotnet/roslyn/issues/17612
```

Comment:
Why should `Error M() { ... }` emit without errors if the signature has an unrecognized type?